src/components/learner-credit-management/data/utils.js

```diff
@@ -551,3 +555,98 @@ export const isLmsBudget = (
+ *
+ * @param courseRuns
+ * @param subsidyExpirationDatetime
+ * @param isLateRedemptionAllowed
+ * @returns {*}
+ */
+export const getAssignableCourseRuns = ({ courseRuns, subsidyExpirationDatetime, isLateRedemptionAllowed }) => {
+  const clonedCourseRuns = courseRuns.map(courseRun => ({
+    ...courseRun,
+    enrollBy: dayjs.unix(courseRun.enrollBy).toISOString(),
+    upgradeDeadline: dayjs.unix(courseRun.upgradeDeadline).toISOString(),
+  }));
+  const assignableCourseRunsFilter = ({ enrollBy, isActive }) => {
+    const enrollByDateThreshold = dayjs(enrollBy).isBefore(
+      Math.max(
+        dayjs(subsidyExpirationDatetime).subtract(MAX_ALLOWABLE_REFUND_THRESHOLD_DAYS, 'days').toDate(),
+        dayjs().add(DAYS_UNTIL_ASSIGNMENT_ALLOCATION_EXPIRATION, 'days').toDate(),
+      ),
+    );
+    if (isLateRedemptionAllowed) {
+      return enrollByDateThreshold;
+    }
+    return isActive && enrollByDateThreshold;
+  };
+  const assignableCourseRuns = clonedCourseRuns.filter(assignableCourseRunsFilter);
+  const sortedAssignableCourseRuns = assignableCourseRuns.sort((a, b) => a.enrollBy - b.enrollBy);
+  return sortedAssignableCourseRuns;
+};
+
+export const isCourseSelfPaced = ({ pacingType }) => pacingType === COURSE_PACING_MAP.SELF_PACED;
+
+export const hasTimeToComplete = ({ end, weeksToComplete }) => {
+  if (!weeksToComplete || !end) {
+    return true;
+  }
+  const today = dayjs();
+  const differenceInWeeks = dayjs(end).diff(today, 'week');
+  return weeksToComplete <= differenceInWeeks;
+};
+
+const isWithinMinimumStartDateThreshold = ({ start }) => dayjs(start).isBefore(dayjs().subtract(START_DATE_DEFAULT_TO_TODAY_THRESHOLD_DAYS, 'days'));
+
+/**
+ * If the start date of the course is before today offset by the START_DATE_DEFAULT_TO_TODAY_THRESHOLD_DAYS
+ * then return today's formatted date. Otherwise, pass-through the start date in ISO format.
+ *
+ * For cases where a start date does not exist, just return today's date.
+ *
+ * @param {string} - start
+ * @param {string} - pacingType
+ * @param {string} - end
+ * @param {number} - weeksToComplete
+ * @returns {string}
+ */
+export const getNormalizedStartDate = ({
+  start, pacingType, end, weeksToComplete,
+}) => {
+  const todayToIso = dayjs().toISOString();
+  if (!start) {
+    return todayToIso;
+  }
+  const startDateIso = dayjs(start).toISOString();
+  if (isCourseSelfPaced({ pacingType })) {
+    if (hasTimeToComplete({ end, weeksToComplete }) || isWithinMinimumStartDateThreshold({ start })) {
+      // always today's date (incentives enrollment)
+      return todayToIso;
+    }
+    return startDateIso;
+  }
+  return startDateIso;
+};
+
+export const getNormalizedEnrollByDate = ({ enrollBy }) => {
+  if (!enrollBy) {
+    return null;
+  }
+  const ninetyDayAllocationOffset = dayjs().add(DAYS_UNTIL_ASSIGNMENT_ALLOCATION_EXPIRATION, 'days');
+  if (dayjs(enrollBy).isAfter(ninetyDayAllocationOffset)) {
+    return ninetyDayAllocationOffset.toISOString();
+  }
```

Comment:
Nit: The variable is suffixed with "Offset" but it is not an offset, it is a date. Maybe "ninetyDaysFromNow" instead?